lib/node-token-validation/src/config/TokenValidationParameters.ts

```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { JwtConstants, TokenType } from "../utils/Constants";
+import { JWK } from "jose";
+
+export type TokenValidationParameters = {
+    // Valid claims
+    validIssuers: Array<string>,
+    validAudiences: Array<string>,
+    validAlgorithms?: Array<string>,
+    validTypes?: Array<string>,
+    // Signature validation
+    issuerSigningKeys?: Array<JWK>,
+    issuerSigningJwksUri?: string,
+    // Other
+    requireExpirationTime?: Boolean,
+    requireSignedTokens?: Boolean,
+    subject?: string
+    nonce?: string,
+    code?: string,
+    accessTokenForAtHash?: string,
+};
+
+export type ValidationParameters = {
```

Comment:
Is it possible to use `Pick` or `Omit` and have `TokenValidationParameters` and `ValidationParameters` share properties? Maintaining these separately might be add on complexity.

Reply:
That's a fair point. I used Partial to add it for now, but if there ends up being more parameters needed I may separate it out to different files.